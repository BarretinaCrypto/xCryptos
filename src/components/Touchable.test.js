import React from 'react';
import renderer from 'react-native-test-utils';

jest.mock('TouchableOpacity', () => 'TouchableOpacity');
jest.mock('TouchableNativeFeedback', () => 'TouchableNativeFeedback');

let Touchable;

describe('when Platorm.OS is ios', () => {
  beforeEach(() => {
    jest.resetModules();

    const reactNative = require('react-native');
    reactNative.Platform.OS = 'ios';

    Touchable = require.requireActual('./Touchable').default;
  });

  it('renders a TouchableOpacity component', () => {
    const tree = renderer(<Touchable />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('when Platorm.OS is not ios', () => {
  beforeEach(() => {
    jest.resetModules();

    const reactNative = require('react-native');
    reactNative.Platform.OS = 'android';

    Touchable = require.requireActual('./Touchable').default;
  });

  it('renders a TouchableNativeFeedback component', () => {
    const tree = renderer(<Touchable />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

