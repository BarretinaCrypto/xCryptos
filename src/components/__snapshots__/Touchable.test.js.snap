// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Platorm.OS is ios renders a TouchableOpacity component 1`] = `<TouchableOpacity />`;

exports[`when Platorm.OS is not ios renders a TouchableNativeFeedback component 1`] = `<TouchableNativeFeedback />`;
